pkg/cmd/admin/network/project_options.go

```diff
@@ -109,83 +97,82 @@ func (p *ProjectOptions) GetProjects() ([]*api.Project, error) {
 	}
 
 	errList := []error{}
-	projectList := []*api.Project{}
+	infoList := []*resource.Info{}
 	_ = r.Visit(func(info *resource.Info, err error) error {
 		if err != nil {
 			return err
 		}
-		project, ok := info.Object.(*api.Project)
+		_, ok := info.Object.(*kapi.Namespace)
 		if !ok {
-			err := fmt.Errorf("cannot convert input to Project: %v", reflect.TypeOf(info.Object))
+			err := fmt.Errorf("cannot convert input to Namespace: %v", reflect.TypeOf(info.Object))
 			errList = append(errList, err)
-			// Don't bail out if one project fails
+			// Don't bail out if one namespace fails
 			return nil
 		}
-		projectList = append(projectList, project)
+		infoList = append(infoList, info)
 		return nil
 	})
 	if len(errList) != 0 {
-		return projectList, kerrors.NewAggregate(errList)
+		return infoList, kerrors.NewAggregate(errList)
 	}
 
-	if len(projectList) == 0 {
-		return projectList, fmt.Errorf("No projects found")
+	if len(infoList) == 0 {
+		return infoList, fmt.Errorf("No projects found")
 	} else {
 		givenProjectNames := sets.NewString(p.ProjectNames...)
 		foundProjectNames := sets.String{}
-		for _, project := range projectList {
-			foundProjectNames.Insert(project.ObjectMeta.Name)
+		for _, info := range infoList {
+			ns, _ := info.Object.(*kapi.Namespace)
+			foundProjectNames.Insert(ns.ObjectMeta.Name)
 		}
 		skippedProjectNames := givenProjectNames.Difference(foundProjectNames)
 		if skippedProjectNames.Len() > 0 {
-			return projectList, fmt.Errorf("Projects %v not found", strings.Join(skippedProjectNames.List(), ", "))
+			return infoList, fmt.Errorf("Projects %v not found", strings.Join(skippedProjectNames.List(), ", "))
 		}
 	}
-	return projectList, nil
+	return infoList, nil
 }
 
-func (p *ProjectOptions) GetNetNamespaces() (*sdnapi.NetNamespaceList, error) {
-	netNamespaces, err := p.Oclient.NetNamespaces().List(kapi.ListOptions{})
```

Comment:
I think for this specific case, given that it is an admin level task and a
specific command that is infrequently called, and the goal is to simplify
and strip down these APIs more generally, that we can drop the support.
However, we should at minimum handle the error.

On Thu, Jun 2, 2016 at 4:56 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/admin/network/project_options.go
> <https://github.com/openshift/openshift-sdn/pull/303#discussion_r65616848>
> :
>
> >  }
> >
> > -func (p *ProjectOptions) GetNetNamespaces() (*sdnapi.NetNamespaceList, error) {
> > -	netNamespaces, err := p.Oclient.NetNamespaces().List(kapi.ListOptions{})
>
> we do a lot of other work to make new clients work with old servers, at
> least for N versions. @smarterclayton <https://github.com/smarterclayton>,
> what were you expecting here?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-sdn/pull/303/files/2c9eb049e3eb48c39db8e863cd2b4460d7a09fec#r65616848>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p330Y2LoiW2P_4X4X6KD_Kp4csjcks5qH0OHgaJpZM4ITCfK>
> .
>